### PR TITLE
Remove stdout/stderr headers when there is nothing in it

### PR DESCRIPTION
**Pull request by linzhp**
When a test failed and there nothing in stdout/stderror, buck should not print out headers with empty body like the one below:

```
====STANDARD OUT====

====STANDARD ERR====

```
This PR fixed that.

### CODE REVIEW

File: src/com/facebook/buck/event/listener/TestResultFormatter.java [hunk 1]
```diff
@@ -218,11 +218,15 @@ public void reportResultSummary(
...
+    if (summaryVerbosity.getIncludeStdOut()
+        && testResult.getStdOut() != null
+        && testResult.getStdOut().length() > 0) {
       addTo.add("====STANDARD OUT====", testResult.getStdOut());
     }
 
-    if (summaryVerbosity.getIncludeStdErr() && testResult.getStdErr() != null) {
+    if (summaryVerbosity.getIncludeStdErr()
+        && testResult.getStdErr() != null
+        && testResult.getStdErr().length() > 0) {
```
Comment (styurin) [inline src/com/facebook/buck/event/listener/TestResultFormatter.java]: The same

File: src/com/facebook/buck/event/listener/TestResultFormatter.java [hunk 2]
```diff
@@ -218,11 +218,15 @@ public void reportResultSummary(
...
-    if (summaryVerbosity.getIncludeStdOut() && testResult.getStdOut() != null) {
+    if (summaryVerbosity.getIncludeStdOut()
+        && testResult.getStdOut() != null
+        && testResult.getStdOut().length() > 0) {
       addTo.add("====STANDARD OUT====", testResult.getStdOut());
     }
 
-    if (summaryVerbosity.getIncludeStdErr() && testResult.getStdErr() != null) {
+    if (summaryVerbosity.getIncludeStdErr()
+        && testResult.getStdErr() != null
```
Comment (ttsugriy) [inline src/com/facebook/buck/event/listener/TestResultFormatter.java]: you can simplify by using [Strings.isNullOrEmpty](https://google.github.io/guava/releases/23.0/api/docs/com/google/common/base/Strings.html#isNullOrEmpty-java.lang.String-)

File: src/com/facebook/buck/event/listener/TestResultFormatter.java [hunk 3]
```diff
@@ -218,11 +218,15 @@ public void reportResultSummary(
       }
     }
 
-    if (summaryVerbosity.getIncludeStdOut() && testResult.getStdOut() != null) {
+    if (summaryVerbosity.getIncludeStdOut()
+        && testResult.getStdOut() != null
+        && testResult.getStdOut().length() > 0) {
```
Comment (styurin) [inline src/com/facebook/buck/event/listener/TestResultFormatter.java]:
```
if (summaryVerbosity.getIncludeStdOut()
    && !Strings.isNullOrEmpty(testResult.getStdOut())) {
```

**Comment by linzhp** [comment]
@ttsugriy Any more concerns about this PR?

File: src/com/facebook/buck/event/listener/TestResultFormatter.java [hunk 4]
```diff
@@ -115,6 +116,7 @@ public void runStarted(
   public void reportResult(ImmutableList.Builder<String> addTo, TestResults results) {
     if (verbosity.shouldPrintBinaryRunInformation() && results.getTotalNumberOfTests() > 1) {
       addTo.add("");
+
```
Comment (bobyangyf) [inline src/com/facebook/buck/event/listener/TestResultFormatter.java]: no need to add extra line

File: src/com/facebook/buck/features/go/GoTest.java [hunk 5]
```diff
@@ -229,7 +229,7 @@ public GoTest(
                   "",
                   Joiner.on(System.lineSeparator()).join(stackTrace),
                   Joiner.on(System.lineSeparator()).join(stdout),
-                  ""));
+                  null));
```
Comment (bobyangyf) [inline src/com/facebook/buck/features/go/GoTest.java]: I don't think we need to make this null. Leave it as is before

**Pull request closed without merging**